mongo/integration/sdam_prose_test.go

```diff
@@ -232,4 +233,36 @@ func TestServerHeartbeatStartedEvent(t *testing.T) {
 		}
 		assert.Equal(t, expectedEvents, actualEvents)
 	})
+
+	mt := mtest.New(t)
+
+	mt.Run("polling must await frequency", func(mt *mtest.T) {
+		var heartbeatStartedCount atomic.Int64
+		servers := map[string]bool{}
+
+		serverMonitor := &event.ServerMonitor{
+			ServerHeartbeatStarted: func(*event.ServerHeartbeatStartedEvent) {
+				heartbeatStartedCount.Add(1)
+			},
+			TopologyDescriptionChanged: func(evt *event.TopologyDescriptionChangedEvent) {
+				for _, srv := range evt.NewDescription.Servers {
+					servers[srv.Addr.String()] = true
+				}
+			},
+		}
+
+		// Create a client with  heartbeatFrequency=100ms,
+		// serverMonitoringMode=poll. Use SDAM to record the number of times the
+		// a heartbeat is started and the number of servers discovered.
+		mt.ResetClient(options.Client().
+			SetServerMonitor(serverMonitor).
+			SetServerMonitoringMode(options.ServerMonitoringModePoll))
+
+		// Per specifications, minHeartbeatFrequencyMS=500ms. So, within the first
+		// 500ms the heartbeatStartedCount should be LEQ to the number of discovered
+		// servers.
+		time.Sleep(500 * time.Millisecond)
+
+		assert.LessOrEqual(mt, heartbeatStartedCount.Load(), int64(len(servers)))
```

Comment:
We could also make this a strong equality.

Reply:
I think that could cause failures in some cases. Go timers (including tickers) guarantee that a goroutine will continue _no earlier than_ the deadline set. However, there are no strong guarantees for how soon _after_ the deadline the goroutine will continue. As a result, it's possible (though unlikely) that the polling loop will not continue as many times as expected during the test duration.